symengine/derivative.cpp

```diff
@@ -74,10 +74,39 @@ class DiffImplementation
 
     static RCP<const Basic> diff(const Zeta &self, const RCP<const Symbol> &x)
     {
-        // TODO: check if it is differentiated wrt s
-        return mul(mul(mul(minus_one, self.get_s()),
-                       zeta(add(self.get_s(), one), self.get_a())),
-                   self.get_a()->diff(x));
+        RCP<const Basic> m1 = self.get_s()->diff(x);
+        RCP<const Basic> diff
+            = mul(mul(mul(minus_one, self.get_s()),
+                      zeta(add(self.get_s(), one), self.get_a())),
+                  self.get_a()->diff(x));
+        if (eq(*m1, *zero)) {
+            return diff;
+        } else {
+            unsigned count = 0;
+            bool found_x = false;
+            for (const auto &a : self.get_args()) {
+                if (eq(*a, *x)) {
+                    found_x = true;
+                    count++;
+                } else if (count < 2 and neq(*a->diff(x), *zero)) {
+                    count++;
+                }
+            }
+            if (count == 1 and found_x) {
+                return Derivative::create(self.rcp_from_this(), {x});
+            }
```

Comment:
Only the above lines after the `} else {` statement are same as our implementation for `LeviCivta` and `FunctionSymbol`. The rest has been modified to do cater to `Zeta`. Should I instead make a function for these lines?